### DISCUSSION TITLE
How handle media(photo, pdf)  messages from chat ?

### DISCUSSION THREAD

**Discussion by Nurbolaaaat** [def-studio/telegraph#441] [Q&A] [answered]
Hi !

1. I need to handle photo and pdf from user.
2. I should store to my local project.
3. I need to send this files to other chat

**Comment by Nurbolaaaat** [comment]
**Also, one moment !

I can not send jpeg file from project to chat (telgraph_chat_id = 102). File is exists. Symlink is configured. In logs from webhook everything is ok. Please, help !)**

![Снимок экрана от 2023-12-05 00-47-29](https://github.com/defstudio/telegraph/assets/138411350/2695a4e8-7ccb-4401-9a59-2489fa886610)

![Снимок экрана от 2023-12-05 00-46-54](https://github.com/defstudio/telegraph/assets/138411350/a3d7416d-006d-4824-9556-0a5bad28d24d)

![image](https://github.com/defstudio/telegraph/assets/138411350/62555dc2-74f7-4564-94be-57f828fa4eae)

**Reply by fabio-ivona** [reply]
@Nurbolaaaat could you please share the dump of the response returned by the ->send() method?

**Reply by Nurbolaaaat** [reply]
{"ok":false,"error_code":400,"description":"Bad Request: invalid file HTTP URL specified: URL host is empty"}

**Reply by Nurbolaaaat** [reply]
also i have "ok":false,"error_code":400,"description":"Bad Request: failed to get HTTP URL content" error

**Reply by fabio-ivona** [reply]
could you dump it before calling send()?

**Comment by fabio-ivona** [comment, marked as answer]
> 1. I need to handle photo and pdf from user.
> 1. I should store to my local project.

as for this one: starting from a Message DTO

from a webhook

```php
$this->message
```


 or Telegraph::botUpdates() method:

```php
$message = Telegraph::botUpdates()->send()->json('result.0.message');
$message = Message::fromArray($message);
```

you can extract photos (or documents as well)

```php
$photo = collect($message->photos())->last();
```

and then, store it:

```php
Telegraph::store($photo, Storage::path('incoming/photos'), 'new-photo.jpg');
```

> 3. I need to send this files to other chat

for this one, your code is correct (`->photo(...)->send()`)

**Reply by Nurbolaaaat** [reply]
Ooh great ! It works ! Much thanks .
